ev3dev2/motor.py

```diff
@@ -1924,6 +2037,181 @@ def get_speed_steering(self, steering, speed):
         return (left_speed, right_speed)
 
 
+class MoveDifferential(MoveTank):
+    """
+    For all methods speed can be an integer or a SpeedInteger object
+    """
+
+    def __init__(self, left_motor_port, right_motor_port, wheel_class, wheel_distance_mm, desc=None, motor_class=LargeMotor):
+        MoveTank.__init__(self, left_motor_port, right_motor_port, desc, motor_class)
+        self.wheel = wheel_class()
+
+        # The distance between the mid point of the wheels.  You may need to
+        # do some test drives to find the correct value for your robot.  It
+        # is not as simple as measuring the distance between the midpoints of
+        # the two wheels. The weight of the robot, center of gravity, etc come
+        # into play.  You can use utils/move_differential.py to call on_arc_left()
+        # to do some test drives in 1-foot circles.  Tweak wheel_distam_mm until
+        # your robot can drive in a 1-foot circle and stop exactly where it started.
+        self.wheel_distance_mm = wheel_distance_mm
+
+        # The circumference of the circle made if this robot were to rotate in place
+        self.circumference_mm = self.wheel_distance_mm * pi
+
+        self.min_circle_radius_mm = self.wheel_distance_mm / 2
+
+    def on_for_distance(self, speed, distance_mm, brake=True, block=True):
+        """
+        Drive forward distance_mm
```

Comment:
Forward? How do negative speeds/distances interact?

Reply:
I will drop the “forward”. Negative speeds and distances should be fine but I will test to be sure.